### PR TITLE
Sync cluster service information when mounting a searchable snapshot

### PR DESCRIPTION
**Pull request by arteam**
Because of the delay in updating the cluster service info (happens every in 15s), we can have outdated information about the free space on disk in `DiskThresholdDecider` when we check whether it's possible to allocate an index. One possible solution for this would be to try to refresh the cluster service info during the preparation of a new searchable snapshot.

Fixes #85018

### CODE REVIEW

**Comment by arteam** [comment]
@elasticmachine update branch

**Comment by arteam** [comment]
@elasticmachine update branch

**Comment by arteam** [comment]
@elasticmachine update branch

File: x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/action/TransportMountSearchableSnapshotAction.java [hunk 1]
```diff
@@ -250,6 +254,8 @@ protected void masterOperation(
                 dataTierAllocationSetting.get(indexSettings);
             }
 
+            clusterInfoService.syncClusterInfo();
```
Comment (fcofdez) [inline x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/action/TransportMountSearchableSnapshotAction.java:257]: I think this do not fixes the issue, at this point the index is not mounted yet. I think this pass the test as the implementation in `InternalClusterInfoService#syncClusterInfo` triggers an async operation and at the point it refreshes the information the index is already mounted I think?

**Pull request closed without merging**